#! /bin/sh /usr/share/dpatch/dpatch-run
## 50_correct_test.dpatch by Thomas Weber <thomas.weber.mail@gmail.com>
##
## All lines beginning with `## DP:' are a description of the patch.
## DP: Take new test.m from upstream CVS; Fix for #410070

@DPATCH@
diff -urNad octave2.9-2.9.9~/scripts/testfun/test.m octave2.9-2.9.9/scripts/testfun/test.m
--- octave2.9-2.9.9~/scripts/testfun/test.m	2006-07-27 19:16:14.000000000 +0200
+++ octave2.9-2.9.9/scripts/testfun/test.m	2007-02-07 15:39:19.000000000 +0100
@@ -146,12 +146,16 @@
   endif
 
   ## locate the file to test
-  __file = file_in_loadpath (__name);
-  if (isempty (__file)) 
-    __file = file_in_loadpath ([__name, ".m"]);
+  __file = file_in_loadpath (__name, "all");
+  if (isempty (__file))
+    __file = file_in_loadpath ([__name, ".m"], "all");
   endif
   if (isempty (__file))
-    __file = file_in_loadpath ([__name, ".cc"]);
+    __file = file_in_loadpath ([__name, ".cc"], "all");
+  endif
+  if (iscell (__file))
+    ## If repeats, return first in path.
+    __file = __file{1};
   endif
   if (isempty (__file))
     if (__grabdemo)
@@ -479,12 +483,12 @@
   pos = [];
 
   ## Find the end of the name
-  right = min(find(def=='('));
+  right = find(def=='(', 1);
   if isempty(right), return; endif
-  right = max(find(def(1:right-1) != ' '));
+  right = find(def(1:right-1) != ' ', 1, "last");
 
   ## Find the beginning of the name
-  left = max([find(def(1:right)==' '),find(def(1:right)=='=')]);
+  left = max([find(def(1:right)==' ', 1, "last"),find(def(1:right)=='=', 1, "last")]);
   if isempty(left), return; endif
   left++;
 
